File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_AttachRequest.js
import { AttachRequest } from './pdu_jstype_AttachRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduAttachRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { AttachRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_string(src.asset_name);
        writer.write_uint64(src.delta_asset_tick);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { AttachRequest } dst
     * @returns { AttachRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.asset_name = reader.read_string();
        dst.delta_asset_tick = reader.read_uint64();
        return dst;
    }

    /**
     * @param { AttachRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { AttachRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new AttachRequest());
    }
}

File: pdu/python/hako_srv_msgs/pdu_cdr_conv_AttachRequest.py
from .pdu_pytype_AttachRequest import AttachRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_AttachRequest(writer: CdrWriter, src: AttachRequest):
    writer.write_string(src.asset_name)
    writer.write_uint64(src.delta_asset_tick)


def cdr_body_to_py_AttachRequest(reader: CdrReader, dst: AttachRequest):
    dst.asset_name = reader.read_string()
    dst.delta_asset_tick = reader.read_uint64()
    return dst


def py_to_cdr_AttachRequest(src: AttachRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_AttachRequest(writer, src)
    return writer.bytes()


def cdr_to_py_AttachRequest(cdr_payload) -> AttachRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = AttachRequest()
    return cdr_body_to_py_AttachRequest(reader, dst)
